### PR TITLE
Stormpath default spring boot starter

### LINKED ISSUE #281: Stormpath default spring boot starter
Add a default starter that incorporates `stormpath-spring-security-webmvc-spring-boot-starter` and `stormpath-thymeleaf-spring-boot-starter`

### PR DESCRIPTION
**Pull request by dogeared**
Includes:

```
extensions/spring/boot/stormpath-default-spring-boot-starter
examples/spring-boot-default
```

resolves #281

### CODE REVIEW

File: examples/spring-boot-default/src/main/java/tutorial/SpringSecurityWebAppConfig.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * @since 1.0.RC5
+ */
+@Configuration
+@Import(StormpathWebSecurityAutoConfiguration.class)
+public class SpringSecurityWebAppConfig extends WebSecurityConfigurerAdapter {
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+            .authorizeRequests()
```
Comment (lhazlewood) [inline examples/spring-boot-default/src/main/java/tutorial/SpringSecurityWebAppConfig.java]: I think we nee do reverse this, i.e. `http.antMatcher("/restricted") ...` as the suggested example/practice due to global security policies overwriting our default.  @mrioan might have some more information as to why - he's currently looking in to this issue.

Comment (dogeared) [inline examples/spring-boot-default/src/main/java/tutorial/SpringSecurityWebAppConfig.java]: Gotcha. Is this related in any way to moving away from having a final `configure` method that calls `doConfigure` implemented in subclasses? I'd thought that's how we guarded against getting our defaults overwritten.

Comment (mrioan) [inline examples/spring-boot-default/src/main/java/tutorial/SpringSecurityWebAppConfig.java]: @dogeared Please remove the `@Import` line, it is not needed any longer. Furthermore, using it was not SpringBoot idiomatic. I introduced that in an intermediate version and Les made me realize that it was wrong.

Comment (dogeared) [inline examples/spring-boot-default/src/main/java/tutorial/SpringSecurityWebAppConfig.java]: Done.

**Comment by dogeared** [comment]
@lhazlewood @mrioan - I just merged in `spring_security_extension` after pulling in @mrioan's auto config update and the example in this PR works well.

**Pull request merged**